src/gateway/sharding/shard_runner.rs

```diff
@@ -487,14 +467,29 @@ impl ShardRunner {
     }
 
     #[cfg_attr(feature = "tracing_instrument", instrument(skip(self)))]
-    async fn update_manager(&self) {
-        self.manager
-            .update_shard_latency_and_stage(
-                self.shard.shard_info().id,
-                self.shard.latency(),
-                self.shard.stage(),
-            )
-            .await;
+    async fn update_runner_info(&self) {
+        let mut runner_info = self.runner_info.lock().await;
+        runner_info.latency = self.shard.latency();
+        runner_info.stage = self.shard.stage();
+    }
+
+    fn make_context(&self) -> Context {
+        Context::new(
+            Arc::clone(&self.data),
+            self.messenger(),
+            self.shard.shard_info().id,
+            Arc::clone(&self.http),
+            #[cfg(feature = "cache")]
+            Arc::clone(&self.cache),
+        )
+    }
```

Comment:
Why move this?

Reply:
It was kind of out of place in the middle of the method list. It's a small method that is unrelated to direct shard state management.